instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/asyncannotationsupport/AsyncOperationEndStrategies.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.asyncannotationsupport;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/** A global registry of {@link AsyncOperationEndStrategy} implementations. */
+public final class AsyncOperationEndStrategies {
+  private static final List<AsyncOperationEndStrategy> STRATEGIES = new CopyOnWriteArrayList<>();
+
+  static {
+    STRATEGIES.add(Jdk8AsyncOperationEndStrategy.INSTANCE);
+  }
+
+  /** Add the passed {@code strategy} to the registry. */
+  public static void registerStrategy(AsyncOperationEndStrategy strategy) {
+    STRATEGIES.add(requireNonNull(strategy));
+  }
+
+  /** Remove the passed {@code strategy} from the registry. */
+  public static void unregisterStrategy(AsyncOperationEndStrategy strategy) {
+    STRATEGIES.remove(strategy);
+  }
+
+  /** Remove all strategies of type {@code strategyClass} from the registry. */
+  public static void unregisterStrategy(Class<? extends AsyncOperationEndStrategy> strategyClass) {
```

Comment:
Do we need this? It seems to not be symmetrical

Reply:
I believe this was also inherited from `AsyncSpanEndStrategy`.  The strategies were initially singletons so "unregistering" them by instance was sufficient but when they became configurable I wanted a way to unregister by type so that I didn't have to track strategy instances in the instrumentation, which generally has no state.

I'm not arguing for keeping that method or porting it to the new API, just providing context as to why that method was originally added.

Reply:
On the other hand: if those strategies were stores as fields/constants, there would be no need for this method; we could unregister them by reference.